tools/windows/DatadogAgentInstaller/CustomActions/NpmCustomAction.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using Datadog.CustomActions.Extensions;
+using Microsoft.Deployment.WindowsInstaller;
+using Microsoft.Win32;
+
+namespace Datadog.CustomActions;
+
+public class NpmCustomAction
+{
+    private static ActionResult EnsureNpmServiceDepdendency(ISession session)
+    {
+        try
+        {
+            var addLocal = session.Property("ADDLOCAL");
+            session.Log($"ADDLOCAL={addLocal}");
+            using var systemProbeDef = Registry.LocalMachine.OpenSubKey(@"SYSTEM\CurrentControlSet\Services\datadog-system-probe", true);
```

Comment:
Why are we using the Registry APIs instead of the service APIs?

Reply:
Wow I'm amazed there's no .NET service APIs and they force you to use pinvoke.

I don't think Windows is likely to change these keys so it's okay.